File: Reto1y2/Pear/config_file.py
import os
import configparser
import requests

config = configparser.ConfigParser()

def create_config_file():
    ip = public_ip()
    name_directory = 'shared_files'
    proto_path = 'protobufs/service.proto'

    #Create the folder where the files will be saved
    try:
        os.mkdir(name_directory)
    except:
        print('The directory already exists')

    config['config'] = {
        'ip_public': f'{ip}',
        'port_server': '8000',
        'port_grpc': '9998',
        'port_mom': '9997',
        'port_rest': '9996',
        'directory': f'{name_directory}',
        'proto_path': f'{proto_path}'
    }

    #Create the file config.conf
    with open('config.conf', 'w') as archivo:
        config.write(archivo)

def public_ip():
    ip = requests.get("https://api.ipify.org").text
    return ip

def get_config():
    config.read('config.conf')

    return config

def get_ip():
    return get_config()['config']['ip_public']

def get_port_server():
    return int(get_config()['config']['port_server'])

def get_port_grpc():
    return int(get_config()['config']['port_grpc'])

def get_port_mom():
    return int(get_config()['config']['port_mom'])

def get_directory():
    return get_config()['config']['directory']

File: Reto1y2/Bootsp/config_file.js
const fs = require('fs');

const create_config_file = async () => {
    const ip = await get_my_ip();
    const name_directory = 'shared_files'
    const proto_path = 'protobufs/service.proto'

    fs.mkdir(name_directory, (err) => {
        if (err) {
            console.log("The directory already exists");
        }
    });

    const config = {
        'config': {
            "ip": ip,
            'port_server': '8000',
            'port_grpc': '9998',
            'port_mom': '9997',
            'port_rest': '9996',
            'directory': `${name_directory}`,
            'proto_path': `${proto_path}`
        }
    };

    await fs.promises.writeFile('config.json', JSON.stringify(config), (err) => {});
};


const get_my_ip = async () => {
    let ip = await fetch('https://ipinfo.io/json').then(response => (response.json()));
    return ip.ip;
};

const get_config = () => {
    let config = fs.readFileSync('config.json');
    return JSON.parse(config);
}

const get_ip = () => get_config().config.ip;
const get_port_server = () => get_config().config.port_server;
const get_port_grpc = () => get_config().config.port_grpc;
const get_port_mom = () => get_config().config.port_mom;
const get_port_rest = () => get_config().config.port_rest;
const get_directory = () => get_config().config.directory;
const get_proto_path = () => get_config().config.proto_path;

module.exports = { 
    create_config_file, 
    get_ip,
    get_port_server,
    get_port_grpc,
    get_port_mom,
    get_port_rest,
    get_directory,
    get_proto_path
};